libsolutil/JSON.cpp

```diff
@@ -87,8 +87,31 @@ bool parse(Json::CharReaderBuilder& _builder, string const& _input, Json::Value&
 	return reader->parse(_input.c_str(), _input.c_str() + _input.length(), &_json, _errs);
 }
 
+/// Takes a JSON value (@ _json) and removes all its members with value 'null' recursively.
+void removeNullMembersHelper(Json::Value& _json)
+{
+	if (_json.type() == Json::ValueType::arrayValue)
+		for (auto& child: _json)
+			removeNullMembersHelper(child);
+	else if (_json.type() == Json::ValueType::objectValue)
+		for (auto const& key: _json.getMemberNames())
+		{
+			Json::Value& value = _json[key];
+			if (value.isNull())
+				_json.removeMember(key);
+			else
+				removeNullMembersHelper(value);
+		}
+}
+
 } // end anonymous namespace
 
+Json::Value removeNullMembers(Json::Value _json)
+{
+	removeNullMembersHelper(_json);
+	return _json;
```

Comment:
Can return value optimization be done here?

Reply:
`Json::value` has move semantics. So the compiler should do the RVO.

Reply:
Isn't there an exception for function parameters? I think you might need an `std::move` here.

Reply:
I also think it needs `move`

Reply:
@leonardoalt @chriseth I was wrong about NRVO. But a move is done implicitly (at least by gcc), I think.

This is what I get if I add the `move`.

```
solidity/libsolutil/JSON.cpp:112:18: error: redundant move in return statement [-Werror=redundant-move]
  112 |  return std::move(_json);
      |         ~~~~~~~~~^~~~~~~
solidity/libsolutil/JSON.cpp:112:18: note: remove ‘std::move’ call
cc1plus: all warnings being treated as errors
```